pydm/tests/widgets/test_enum_combo_box.py

```diff
@@ -0,0 +1,330 @@
+    if not pydm_enumcombobox._connected:
+        assert "PV is disconnected." in actual_tooltip
+    elif not write_access:
+        if is_pydm_app() and pydm_enumcombobox.app.is_read_only():
+            assert "Running PyDM on Read-Only mode." in actual_tooltip
+        else:
+            assert "Access denied by Channel Access Security." in actual_tooltip
+    elif not pydm_enumcombobox._has_enums:
+        assert "Enums not available" in actual_tooltip
+
+
+@pytest.mark.parametrize("values, selected_index, expected", [
+    (("RUN", "STOP"), 0, "RUN"),
+    (("RUN", "STOP"), 1, "STOP"),
+])
+def test_enum_strings_changed(qtbot, signals, values, selected_index, expected):
+    """
+    Test the widget's handling of enum strings, which are choices presented to the user, and the widget's ability to
+    update the selected enum string when the user provides a choice index.
+
+    Expectations:
+    The widget displays the correct enum string whose index from the enum string tuple is selected by the user.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        pytest-qt window for widget testing
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    values : tuple
+        A set of enum strings for the user to choose from
+    selected_index : int
+        The index from the enum string tuple chosen by the user
+    expected : str
+        The expected enum string displayed by the widget after receiving the user's choice index
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    signals.enum_strings_signal.connect(pydm_enumcombobox.enumStringsChanged)
+    signals.enum_strings_signal.emit(values)
+
+    signals.new_value_signal[type(selected_index)].connect(pydm_enumcombobox.channelValueChanged)
+    signals.new_value_signal[type(selected_index)].emit(selected_index)
+
+    assert pydm_enumcombobox.value == selected_index
+    assert pydm_enumcombobox.currentText() == expected
+
+
+@pytest.mark.parametrize("value", [
+    0,
+    -1,
+    -2,
+    1,
+    2,
+])
+def test_value_changed(qtbot, signals, value):
+    """
+    Test the widget's handling of the value changed event. In this case, it's the enum string index choice change.
+
+    Expectations:
+    The widget records the correct the enum string index change.
+
+    Parameters
+    ----------
+    qtbot : fixture
+       pytest-qt window for widget testing
+    signals : fixture
+       The signals fixture, which provides access signals to be bound to the appropriate slots
+    value : int
+       The index of the enum string to be displayed by the widget
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    signals.new_value_signal[type(value)].connect(pydm_enumcombobox.channelValueChanged)
+    signals.new_value_signal[type(value)].emit(value)
+
+    assert pydm_enumcombobox.value == value
```

Comment:
What is the point of testing this 5 times? Is there an error you expect to catch here?

Reply:
I thought 0, positive, and negative indices should be tested. As for the 25 and -25 index values, that might not be necessary, so I'll remove them.